### PR TITLE
Remove potential sources of flakes for kms_transformation_test.go.

### PR DESCRIPTION
**Pull request by immutableT**
**What this PR does / why we need it**:
Remove potential sources for flakes in TestKMSPlugin test.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #
#60614
**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: test/integration/master/kms_transformation_test.go [hunk 1]
```diff
@@ -139,13 +141,19 @@ func TestKMSProvider(t *testing.T) {
...
 }
 
-func getDEKFromKMSPlugin(pluginMock *base64Plugin) ([]byte, error) {
-	select {
-	case e := <-pluginMock.encryptRequest:
-		return e.Plain, nil
-	case <-time.After(time.Second):
-		return nil, fmt.Errorf("timed-out while getting encryption request from KMS Plugin Mock")
+func getDEKFromKMSPlugin(pluginMock *base64Plugin, serveErrorChan chan error) ([]byte, error) {
+	if len(serveErrorChan) != 0 {
```
Comment (mikedanese) [inline test/integration/master/kms_transformation_test.go]: It seems like you should check this outside the function

Comment (immutableT) [reply test/integration/master/kms_transformation_test.go]: Yes, makes perfect sense.

File: test/integration/master/kms_transformation_test.go [hunk 2]
```diff
@@ -139,13 +146,15 @@ func TestKMSProvider(t *testing.T) {
 	}
 }
 
-func getDEKFromKMSPlugin(pluginMock *base64Plugin) ([]byte, error) {
-	select {
-	case e := <-pluginMock.encryptRequest:
-		return e.Plain, nil
-	case <-time.After(time.Second):
-		return nil, fmt.Errorf("timed-out while getting encryption request from KMS Plugin Mock")
+func getDEKFromKMSPlugin(pluginMock *base64Plugin, serveErrorChan chan error) ([]byte, error) {
```
Comment (liggitt) [inline test/integration/master/kms_transformation_test.go]: why is serveErrorChan passed in?

**Comment by thockin** [comment]
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/60666)

File: test/integration/master/kms_transformation_test.go (hunk 2 above)
Comment (immutableT) [reply test/integration/master/kms_transformation_test.go]: Thanks for catching this - no need.

**Comment by liggitt** [comment]
doesn't really remove any flake paths, but simplifies things and adds more output that will be useful if flakes are encountered.
/lgtm

**Comment by immutableT** [comment]
@liggitt Could you, please, add a milestone so that this could merge.

**Comment by liggitt** [comment]
> @liggitt Could you, please, add a milestone so that this could merge.

since the changes in this PR don't really remove flake paths, I would probably just hold it until code freeze lifts

**Comment by immutableT** [comment]
IMHO, this PR does remove one potential source of a flake - reliance on time.After in getDEKFromKMSPlugin:
case <-time.After(time.Second):
		return nil, fmt.Errorf("timed-out while getting encryption request from KMS Plugin Mock")

However, if things have been stable (after your change), I agree it is better to wait.

**Comment by liggitt** [comment]
> IMHO, this PR does remove one potential source of a flake - reliance on time.After in getDEKFromKMSPlugin:
> case <-time.After(time.Second):
> return nil, fmt.Errorf("timed-out while getting encryption request from KMS Plugin Mock")

you didn't remove that flake source, you simply changed the failure from a timeout to an immediate failure if the encrypt request channel doesn't have a value available to receive.

> However, if things have been stable (after your change), I agree it is better to wait.

I haven't seen any flakes since my fix merged, though I don't understand how the timeout increase would have helped there - https://storage.googleapis.com/k8s-gubernator/triage/index.html?pr=1&test=TestKMSProvider

**Comment by immutableT** [comment]
I don't completely understand this either, but a college of mine suggested that in the scenario below (given the small value of time.After and when running under load):
select {
	case e := <-pluginMock.encryptRequest:
		return e.Plain, nil
	case <-time.After(time.Microsecond):
		return nil, fmt.Errorf("timed-out while getting encryption request from KMS Plugin Mock")
	}
It may be possible for the second case to evaluate to ready before or at the same time as the first case (even though there is something in the queue). Hence, the reason for getting rid of the select.

**Comment by liggitt** [comment]
> It may be possible for the second case to evaluate to ready before or at the same time as the first case (even though there is something in the queue). Hence, the reason for getting rid of the select.

that's a good point... this is still worth merging, though absent any flakes, I'd still hold this until freeze lifts

**Comment by immutableT** [comment]
Sounds good - thanks for the review.

**Pull request merged**